pennylane_lightning_kokkos/lightning_kokkos.py

```diff
@@ -83,8 +147,21 @@ class LightningKokkos(LightningQubit):
         "Identity",
     }
 
-    def __init__(self, wires, *, sync=True, c_dtype=np.complex128, shots=None, batch_obs=False):
-        super().__init__(wires, c_dtype=c_dtype, shots=shots)
+    def __init__(
+        self, wires, *, sync=True, c_dtype=np.complex128, shots=None, batch_obs=False, analytic=None
+    ):
+        if c_dtype is np.complex64:
+            r_dtype = np.float32
+            self.use_csingle = True
+        elif c_dtype is np.complex128:
+            r_dtype = np.float64
+            self.use_csingle = False
+        else:
+            raise TypeError(f"Unsupported complex Type: {c_dtype}")
+        super().__init__(wires, shots=shots, r_dtype=r_dtype, c_dtype=c_dtype, analytic=analytic)
+        self._state = self._create_basis_state(0)
```

Comment:
Won't this break the device if we are on the GPU instead of the CPU? Do we need to specialise the state creation to handle CPU or GPU memory, and offload to the appropriate type based on the backend?

Reply:
let me add it.

Reply:
done